src/Symfony/Component/BrowserKit/Tests/ClientTest.php

```diff
@@ -620,6 +620,31 @@ public function testSetServerParameter()
         $this->assertEquals('testua', $client->getServerParameter('HTTP_USER_AGENT'));
     }
 
+    public function testUnSetServerParameter()
+    {
+        $client = new TestClient();
+
+        $this->assertEquals('', $client->getServerParameter('HTTP_HOST'));
+        $this->assertEquals('Symfony BrowserKit', $client->getServerParameter('HTTP_USER_AGENT'));
+
+        $client->setServerParameter('HTTP_HOST', 'testhost');
+        $client->unSetServerParameter('HTTP_HOST');
+
+        $this->assertEquals('', $client->getServerParameter('HTTP_HOST'));
+
+        $client->setServerParameter('HTTP_USER_AGENT', 'testua');
+        $client->unSetServerParameter('HTTP_USER_AGENT');
+
+        $this->assertEquals('', $client->getServerParameter('HTTP_USER_AGENT'));
```

Comment:
I think you should try executing a request and test `$client->getInternalRequest()->getServer()` instead, because it's not testing the header has been unset.